tools/codegen/gen_functionalization_type.py

```diff
@@ -371,43 +379,68 @@ def emit_inplace_functionalization_body(
         {return_type} tmp_output;
         {{
           at::AutoDispatchSkipFunctionalize guard;
-          // The functionalization pass explicitly doesn't pass out= parameters to the redispatch
           {functional_call_str}
         }}
         {mutable_input_post_processing}
         {return_str(f)};
       }}
     }}"""
 
+# Generates the FunctionalizeAddBackViews kernels for:
+# - view_copy ops (which get turned back into view ops)
+def emit_view_copy_to_view_kernel(g: NativeFunctionsViewGroup) -> str:
+    view_sig = DispatcherSignature(g.view.func)
+    assert g.view_copy is not None
+    view_copy_sig = DispatcherSignature(g.view_copy.func)
 
-def emit_declaration_for_noncomposite_views(f: NativeFunction) -> str:
-    # For every view op, we need a corresponding "inverse view" function.
-    # This generates the declarations so we get a good compiler error when someone adds a new view.
-    view_inverse_sig = ViewInverseSignature(f)
-    return view_inverse_sig.decl()
+    view_api_name = g.view.func.name.unambiguous_name()
+    exprs = ', '.join([e.expr for e in translate(view_copy_sig.arguments(), view_sig.arguments())])
```

Comment:
Sometime we should add some APIs that make doing this kind of "call-to-call" less wordy haha